files/grest/rpc/blockchain/txs_for_addr.sql

```diff
@@ -0,0 +1,90 @@
+DROP FUNCTION IF EXISTS grest.addr_txs (text, uinteger, character varying);
+
+CREATE FUNCTION grest.addr_txs (_address text, _limit uinteger DEFAULT 1000, _orderby character varying DEFAULT 'desc' ::character varying)
+    RETURNS TABLE (
+        in_addrs jsonb,
+        out_addrs jsonb,
+        tx_hash text,
+        block_idx uinteger,
+        block_hash text,
+        epoch_no uinteger,
+        epoch_slot_no uinteger,
+        block_time timestamp without time zone,
+        stake_addr character varying,
+        is_withdrawal text,
+        is_delegation text,
+        script_data_hash text)
+    LANGUAGE plpgsql
+    STABLE
+    AS $$
+BEGIN
+    RETURN QUERY (
+        SELECT
+            x.in_addrs, x.out_addrs, encode(txhash::bytea, 'hex') AS tx_hash, x.block_idx, encode(bhash::bytea, 'hex') AS block_hash, x.epoch_no, x.epoch_slot_no, x.block_time, coalesce(stakeDel, stakeWith) AS stake_addr, x.is_withdrawal, x.is_delegation, 'TODO' AS script_data_hash FROM (
+            SELECT
+                (
+                    SELECT
+                        jsonb_agg(txo.address)
+                    FROM tx_out txo, tx_in txi
+                WHERE
+                    txo.index = txi.tx_out_index
+                    AND txo.tx_id = txi.tx_out_id
+                    AND txi.tx_in_id = txb.id) in_addrs, (
+                SELECT
+                    jsonb_agg(txo.address) AS addr_list FROM tx_out txo
+            WHERE
+                txo.tx_id = txb.id GROUP BY txo.tx_id) out_addrs, CASE WHEN EXISTS (
+                SELECT
+                    NULL FROM delegation d
```

Comment:
Could this be an issue if multiple addresses are in the transaction and it's a delegation by the address we are not interested in?

Might need to join on `addr_id` as well here. Same for withdrawals.

Reply:
I think a 'yes' / 'no' is still valid at transaction level imo